frontera/contrib/messagebus/zeromq/socket_config.py

```diff
@@ -1,29 +1,33 @@
 # -*- coding: utf-8 -*-
+from socket import getaddrinfo, gaierror
 
 
 class SocketConfig(object):
-
-    hostname = None
-    base_port = None
-
     def __init__(self, hostname, base_port):
-        self.hostname = hostname
-        self.base_port = base_port
+        try:
+            addr_tuple = getaddrinfo(hostname, base_port)[0][4]
+            self._ip_addr = addr_tuple[0]
+            self._base_port = addr_tuple[1]
+        except gaierror:
+            raise gaierror("Hostname '%s' could not be resolved" % hostname)
+
+    def is_ipv6(self):
+        return self._ip_addr.rfind(':')
```

Comment:
https://docs.python.org/2/library/socket.html?highlight=getaddrinfo#socket.getaddrinfo
 >(a (address, port) 2-tuple for AF_INET, a (address, port, flow info, scope id) 4-tuple for AF_INET6)